crates/orchestrator/src/tests/jobs/mod.rs

```diff
@@ -501,3 +507,90 @@ fn build_job_item_by_type_and_status(job_type: JobType, job_status: JobStatus, i
         version: 0,
     }
 }
+
+#[rstest]
+#[case(JobType::SnosRun, JobStatus::Failed)]
+#[tokio::test]
+async fn handle_job_failure_with_failed_job_status_works(#[case] job_type: JobType, #[case] job_status: JobStatus) {
+    TestConfigBuilder::new().build().await;
```

Comment:
As mentioned in [here](https://github.com/madara-alliance/madara-orchestrator/pull/61#discussion_r1709195363
).

`TestConfigBuilder` allows for customisation over any `external client` of our choice,
moving it to a global fixture is not feasible since all test cases have different customisation requirements.

We can make fixture for tests under same scope if they require same customised external clients.

for eg : 
All tests under `da_job` if require same config customisation, can implement a `fixture` just for themselves.
similarly for other scopes .

We can create a separate issue for this and resolve there.

Reply:
sounds good to me